rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java

```diff
@@ -1729,6 +1730,16 @@ private JavaType type(@Nullable com.sun.tools.javac.code.Type type, List<Symbol>
                     if (sym.owner instanceof Symbol.ClassSymbol) {
                         owner = TypeUtils.asFullyQualified(type(sym.owner.type, stackWithSym));
                     }
+
+
+                    // Add annotation information to JavaType.Class
+                    List<JavaType.Class> annotations = new ArrayList<>();
+                    if(sym.getMetadata() != null && sym.getMetadata().getDeclarationAttributes() != null ) {
+                        //  TODO: create type by recursive call (see 1700) ?!
```

Comment:
Yeah, you will want to recursively call type(), it can be easy to get into a cycle here, that is why we have a stack of symbols already visited. 
something like : `TypeUtils.asFullyQualified(type(a.type, stackWithSym));`